File: src/Animal.java
public class Animal {
    private static int count = 0;
    private long id;
    private String typeAnimal;
    private String birthdate;
    private String arrivalDate;
    private double weight;
    private String[] diseases ;
    private String feedingFrequency;
    private String feedingType;
    private boolean vaccines;

    public Animal(String typeAnimal, String birthdate, String arrivalDate, Double weight, String[] diseases, String feedingFrequency, String feedingType, Boolean vaccines) {
        this.typeAnimal = typeAnimal;
        this.birthdate = birthdate;
        this.arrivalDate = arrivalDate;
        this.weight = weight;
        this.diseases = diseases;
        this.feedingFrequency = feedingFrequency;
        this.feedingType = feedingType;
        this.vaccines = vaccines;
        this.id = count;
        count ++;
    }

    public Animal() {

    }

    public String getTypeAnimal() {
        return typeAnimal;
    }

    public String getBirthdate() {
        return birthdate;
    }

    public String getArrivalDate() {
        return arrivalDate;
    }

    public double getWeight() {
        return weight;
    }
    public void setWeight(double weight) {
        this.weight = weight;
    }

    public String[] getDiseases() {
        return diseases;
    }

    public String getFeedingFrequency() {
        return feedingFrequency;
    }

    public String getFeedingType() {
        return feedingType;
    }

    public boolean isVaccines() {
        return vaccines;
    }

    public long getId(){
        return this.id;
    }
}

File: src/Animal.py
class Animal:
    __count = 1
    __id = 0
    __type_animal = ""
    __birthdate = ""
    __arrival_date = ""
    __weight = 0
    __diseases = []
    __feeding_frequency = ""
    __feeding_type = ""
    __vaccines = True

    def __init__(self, type_animal, birthdate, arrival_date, weight, diseases, feeding_frequency, feeding_type,
                 vaccines):
        self.__type_animal = type_animal
        self.__birthdate = birthdate
        self.__arrival_date = arrival_date
        self.__weight = weight
        self.__diseases = diseases
        self.__feeding_frequency = feeding_frequency
        self.__feeding_type = feeding_type
        self.__vaccines = vaccines
        self.__id = Animal.__count
        Animal.__count += 1

    def get_type_animal(self):
        return self.__type_animal

    def get_birthdate(self):
        return self.__birthdate

    def get_arrival_date(self):
        return self.__arrival_date

    def get_weight(self):
        return self.__weight

    def set_weight(self, weight):
        self.__weight = weight

    def get_diseases(self):
        return self.__diseases

    def get_feeding_frequency(self):
        return self.__feeding_frequency

    def get_feeding_type(self):
        return self.__feeding_type

    def get_vaccines(self):
        return self.__vaccines

    def get_id(self):
        return self.__id
